backend/hubspot/hubspot.go

```diff
@@ -373,8 +366,41 @@ func (h *HubspotApi) CreateCompanyForWorkspaceImpl(ctx context.Context, workspac
 				Name:     "domain",
 				Value:    domain,
 			},
+			{
+				Property: "hex_link",
+				Name:     "hex_link",
+				Value:    hexLink,
+			},
 		},
-	})
+	}
+
+	if companyID, err = pollHubspot(func() (*int, error) {
+		id, err := h.getCompany(ctx, name)
+		if err != nil {
+			return nil, err
+		}
+
+		log.WithContext(ctx).
+			WithField("name", name).
+			Warnf("company already exists in Hubspot. updating")
+
+		if id != nil {
+			_, err := h.hubspotClient.Companies().Update(*id, companyProperties)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		return id, nil
+	}); companyID != nil {
+		return
```

Comment:
sorry, should have been a bit more precise here. i think the new logic should go out here to happen once, when the company is found. the inline function above is used to actually find the company in a loop until it appears

Reply:
Oh, right! My bad.